base/hashing.jl

```diff
@@ -55,7 +55,6 @@ end
 
 ## symbol & expression hashing ##
 
-hash(x::Symbol, h::UInt) = 3*object_id(x) - h
```

Comment:
I imagine it was either a #265-related bug (hash was pretty likely to run into bugs of that sort because of the fallback definition), or an oversight. `object_id` is also not specialized (and should just end up always inlined too)